modules/core/src/views/view-manager.js

```diff
@@ -161,9 +161,21 @@ export default class ViewManager {
       this._setSize(props.width, props.height);
     }
 
-    this._rebuildViewportsFromViews();
+    if (!this._isUpdating) {
+      this._update();
+    }
+  }
+
+  _update() {
+    this._isUpdating = true;
+    while (this._needsUpdate) {
```

Comment:
This loop looks a bit scary...

Do we only need to do the update twice, or can we actually get into a loop here? Can there be a cascading change? Do we want to bail at some point if that happens?

Alternatively, could we trigger all the controllers first so we don't get a loop?

Even if we don't do a better implementation now, maybe leave a comment as to what our options are?

Reply:
The controllers must be updated twice for a viewport transition.

1. User initiates a fly-to transition: controller updated with destination viewport props
2. Transition start invokes `onViewStateChange`: controller updated with in-transition viewport props

I can use two ifs.